tests/test_shellcodes.py

```diff
@@ -0,0 +1,88 @@
+import random
+import re
+
+import archinfo
+import angr
+
+import pwnlib
+
+from angr.procedures.definitions import SimLibrary
+from angr.simos import SimLinux
+from archinfo import Endness
+from rex.exploit import Shellcodes
+
+
+arch_to_pwntools = {
+    'ARMEL': 'arm',
+    'MIPS32': 'mips',
+    'X86': 'i386',
+    'AMD64': 'amd64',
+}
+endness_to_pwntools = {
+    Endness.LE: 'little',
+    Endness.BE: 'big'
+}
+
+def test_dupsh(arch, fd_to_dup):
```

Comment:
this doesn't work with nose, does it? Generally, the way we parameterize tests like this is by using nose's iterator test feature. Check out [this](https://github.com/angr/angr/blob/master/tests/test_callable.py) for an example.

Reply:
fixed in 61c406a